### PR TITLE
P1766R1 Mitigating minor modules maladies

### LINKED ISSUE #2992: [2019-07 CWG Motion 13] P1766R1 Mitigating minor modules maladies

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #2992.

### CODE REVIEW

File: source/declarations.tex [hunk 1]
```diff
@@ -735,6 +735,23 @@
 typedef decltype([]{}) C;       // the closure type has no name for linkage purposes
 \end{codeblock}
 \end{example}
```
Comment (burblebee) [inline source/declarations.tex]: I think if you add a newline after this example, the next sentence will start out on a new line and look better.

File: source/declarations.tex [hunk 2]
```diff
@@ -735,6 +735,23 @@
...
+An unnamed class with a typedef name for linkage purposes shall not
+\begin{itemize}
+\item
+  declare any members
+  other than non-static data members, member enumerations, or member classes,
+\item
+  have any base classes or default member initializers, or
+  contain a lambda-expression,
```
Comment (burblebee) [inline source/declarations.tex]: Missing `\item` before "contain a lambda-expression" and, lambda-expression should be \grammarterm'ed.

File: source/compatibility.tex [hunk 3]
```diff
@@ -1922,6 +1922,34 @@
...
 \rSec2[diff.cpp17.dcl.dcl]{\ref{dcl.dcl}: declarations}
 
+\diffref{dcl.typedef}
+\change Unnamed classes with a typedef name for linkage purposes
+can only contain C-compatible constructs.
+\rationale Necessary for implementability.
+\effect Valid C++ 2017 code may be ill-formed in this International Standard.
+\begin{codeblock}
+typedef struct {
+  void f() {}           // ill-formed; previously well-formed
```
Comment (burblebee) [inline source/compatibility.tex]: FYI - our coding style guidelines say that the starting columns for comments should be multiples of 4, but other examples' comments in [diff.cpp17.dcl.dcl] start on column 25 so I wouldn't change this without changing the others as well.

File: source/compatibility.tex [hunk 4]
```diff
@@ -1922,6 +1922,34 @@
...
+with no diagnostic required.
+\begin{codeblock}
+// Translation unit 1
+int f(int a = 42);
+int g() { return f(); }
+
+// Translation unit 2
+int f(int a = 76) { return a; } // ill-formed (no diagnostic required); previously well-formed
+int g();
+int main() { return g(); }      // used to return 42
```
Comment (burblebee) [inline source/compatibility.tex]: ... however, these comments should be moved to multiples of 4 to follow our  guidelines.

File: source/declarations.tex (hunk 2 above)
Comment (jensmaurer) [reply source/declarations.tex]: Fixed.

File: source/declarations.tex (hunk 1 above)
Comment (jensmaurer) [reply source/declarations.tex]: Fixed.

File: source/compatibility.tex (hunk 3 above)
Comment (jensmaurer) [reply source/compatibility.tex]: We seem to be measuring differently. This is column 24 in my book.

File: source/compatibility.tex (hunk 4 above)
Comment (jensmaurer) [reply source/compatibility.tex]: These are at col 32. I've moved them to col 40 to satisfy the "at least two spaces" rule.

File: source/compatibility.tex (hunk 3 above)
Comment (burblebee) [reply source/compatibility.tex]: Eek!  You're right!  Vim has been lying to me!  Which means all of *my* comments are off!!!
```
  void f() {}           // ill-formed; previously well-formed
0         1         2
0123456780912345678091234
...
compatibility.tex [+]                                         1932,25        77%
```
The "1932,25" above is what vim thinks is the line and column.

Comment (jensmaurer) [reply source/compatibility.tex]: I'd guess vim starts counting at 1, but everybody else starts counting at 0. I'm not sure who's right. What would a traditional typewriter do if you hit the "tab" key?

Comment (burblebee) [reply source/compatibility.tex]: Opened #3131 for this.

**Pull request merged**